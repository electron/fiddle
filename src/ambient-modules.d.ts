declare module 'namor';
declare module 'shell-path';
