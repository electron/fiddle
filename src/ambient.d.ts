import * as MonacoType from 'monaco-editor';

import { App } from './renderer/app';

declare global {
  interface Window {
    ElectronFiddle: {
      app: App;
      appPaths: Record<string, string>;
      monaco: typeof MonacoType;
    };
  }
}
