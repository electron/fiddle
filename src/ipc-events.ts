export enum IpcEvents {
  OPEN_SETTINGS = 'OPEN_SETTINGS',
  LOAD_GIST_REQUEST = 'LOAD_GIST_REQUEST',
  LOAD_ELECTRON_EXAMPLE_REQUEST = 'LOAD_ELECTRON_EXAMPLE_REQUEST',
  FIDDLE_RUN = 'FIDDLE_RUN',
  FIDDLE_PACKAGE = 'FIDDLE_PACKAGE',
  FIDDLE_MAKE = 'FIDDLE_MAKE',
  MONACO_EXECUTE_COMMAND = 'MONACO_EXECUTE_COMMAND',
  MONACO_TOGGLE_OPTION = 'MONACO_TOGGLE_OPTION',
  FS_NEW_FIDDLE = 'FS_NEW_FIDDLE',
  FS_NEW_TEST = 'FS_NEW_TEST',
  FS_OPEN_FIDDLE = 'FS_OPEN_FIDDLE',
  FS_OPEN_TEMPLATE = 'FS_OPEN_TEMPLATE',
  FS_SAVE_FIDDLE = 'FS_SAVE_FIDDLE',
  FS_SAVE_FIDDLE_AS = 'FS_SAVE_FIDDLE_AS',
  FS_SAVE_FIDDLE_GIST = 'FS_SAVE_FIDDLE_GIST',
  FS_SAVE_FIDDLE_FORGE = 'FS_SAVE_FIDDLE_FORGE',
  FS_SAVE_FIDDLE_DIALOG = 'FS_SAVE_FIDDLE_DIALOG',
  FS_SAVE_FIDDLE_ERROR = 'FS_SAVE_FIDDLE_ERROR',
  SHOW_WARNING_DIALOG = 'SHOW_WARNING_DIALOG',
  SHOW_CONFIRMATION_DIALOG = 'SHOW_CONFIRMATION_DIALOG',
  SHOW_WELCOME_TOUR = 'SHOW_WELCOME_TOUR',
  CLEAR_CONSOLE = 'CLEAR_CONSOLE',
  LOAD_LOCAL_VERSION_FOLDER = 'LOAD_LOCAL_VERSION_FOLDER',
  SHOW_LOCAL_VERSION_FOLDER_DIALOG = 'SHOW_LOCAL_VERSION_FOLDER_DIALOG',
  BISECT_COMMANDS_TOGGLE = 'BISECT_COMMANDS_TOGGLE',
  SHOW_INACTIVE = 'SHOW_INACTIVE',
  BEFORE_QUIT = 'BEFORE_QUIT',
  CONFIRM_QUIT = 'CONFIRM_QUIT',
  GET_APP_PATHS = 'GET_APP_PATHS',
  SELECT_ALL_IN_EDITOR = 'SELECT_ALL_IN_EDITOR',
  BLOCK_ACCELERATORS = 'BLOCK_ACCELERATORS',
  SET_SHOW_ME_TEMPLATE = 'SET_SHOW_ME_TEMPLATE',
  CLICK_TITLEBAR_MAC = 'CLICK_TITLEBAR_MAC',
  TASK_BISECT = 'TASK_BISECT',
  TASK_TEST = 'TASK_TEST',
  TASK_DONE = 'TASK_DONE',
  OUTPUT_ENTRY = 'OUTPUT_ENTRY',
  RELOAD_WINDOW = 'RELOAD_WINDOW',
}

export const ipcMainEvents = [
  IpcEvents.FS_SAVE_FIDDLE_DIALOG,
  IpcEvents.FS_SAVE_FIDDLE,
  IpcEvents.SHOW_WARNING_DIALOG,
  IpcEvents.SHOW_CONFIRMATION_DIALOG,
  IpcEvents.SHOW_LOCAL_VERSION_FOLDER_DIALOG,
  IpcEvents.SHOW_INACTIVE,
  IpcEvents.CONFIRM_QUIT,
  IpcEvents.SET_SHOW_ME_TEMPLATE,
  IpcEvents.BLOCK_ACCELERATORS,
  IpcEvents.CLICK_TITLEBAR_MAC,
  IpcEvents.OUTPUT_ENTRY,
  IpcEvents.TASK_DONE,
  IpcEvents.RELOAD_WINDOW,
];

export const ipcRendererEvents = [
  IpcEvents.OPEN_SETTINGS,
  IpcEvents.LOAD_GIST_REQUEST,
  IpcEvents.LOAD_ELECTRON_EXAMPLE_REQUEST,
  IpcEvents.FIDDLE_RUN,
  IpcEvents.FIDDLE_PACKAGE,
  IpcEvents.FIDDLE_MAKE,
  IpcEvents.MONACO_EXECUTE_COMMAND,
  IpcEvents.MONACO_TOGGLE_OPTION,
  IpcEvents.FS_NEW_FIDDLE,
  IpcEvents.FS_NEW_TEST,
  IpcEvents.FS_OPEN_FIDDLE,
  IpcEvents.FS_OPEN_TEMPLATE,
  IpcEvents.FS_SAVE_FIDDLE,
  IpcEvents.FS_SAVE_FIDDLE_AS,
  IpcEvents.FS_SAVE_FIDDLE_FORGE,
  IpcEvents.FS_SAVE_FIDDLE_GIST,
  IpcEvents.SHOW_WELCOME_TOUR,
  IpcEvents.CLEAR_CONSOLE,
  IpcEvents.LOAD_LOCAL_VERSION_FOLDER,
  IpcEvents.BISECT_COMMANDS_TOGGLE,
  IpcEvents.BEFORE_QUIT,
  IpcEvents.GET_APP_PATHS,
  IpcEvents.SELECT_ALL_IN_EDITOR,
  IpcEvents.SET_SHOW_ME_TEMPLATE,
  IpcEvents.TASK_BISECT,
  IpcEvents.TASK_TEST,
];

export const WEBCONTENTS_READY_FOR_IPC_SIGNAL =
  'WEBCONTENTS_READY_FOR_IPC_SIGNAL';
