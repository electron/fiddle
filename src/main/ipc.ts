import { EventEmitter } from 'events';

import { ipcMain } from 'electron';

import {
  IpcEvents,
  WEBCONTENTS_READY_FOR_IPC_SIGNAL,
  ipcMainEvents,
} from '../ipc-events';
import { getOrCreateMainWindow } from './windows';

/**
 * The main purpose of this class is to be the central
 * gathering place for IPC calls the main process sends
 * or listens to.
 *
 * @class IpcManager
 * @extends {EventEmitter}
 */
class IpcMainManager extends EventEmitter {
  public readyWebContents = new WeakSet<Electron.WebContents>();
  private messageQueue = new WeakMap<
    Electron.WebContents,
    Array<[IpcEvents, Array<any> | undefined]>
  >();

  constructor() {
    super();

    ipcMainEvents.forEach((name) => {
      ipcMain.removeAllListeners(name);
      ipcMain.on(name, (...args: Array<any>) => this.emit(name, ...args));
    });

    ipcMain.on(
      WEBCONTENTS_READY_FOR_IPC_SIGNAL,
      (event: Electron.IpcMainEvent) => {
        this.readyWebContents.add(event.sender);

        const queue = this.messageQueue.get(event.sender);
        this.messageQueue.delete(event.sender);
        if (!queue) return;
        for (const item of queue) {
          this.send(item[0], item[1], event.sender);
        }
      },
    );
  }

  /**
   * Send an IPC message to an instance of Electron.WebContents.
   * If none is specified, we'll automatically go with the main window.
   *
   * @param {IpcEvents} channel
   * @param {Array<any>} [args]
   * @param {Electron.WebContents} [target]
   */
  public send(
    channel: IpcEvents,
    args?: Array<any>,
    target?: Electron.WebContents,
  ) {
    const _target = target || getOrCreateMainWindow().webContents;
    const _args = args || [];
    if (!this.readyWebContents.has(_target)) {
      const existing = this.messageQueue.get(_target) || [];
      this.messageQueue.set(_target, [...existing, [channel, args]]);
      return;
    }

    _target.isDestroyed() || _target.send(channel, ..._args);
  }

  public handle(
    channel: IpcEvents,
    listener: (event: Electron.IpcMainInvokeEvent, ...args: any[]) => any,
  ) {
    // there can be only one, so remove previous one first
    ipcMain.removeHandler(channel);
    ipcMain.handle(channel, listener);
  }

  public handleOnce(
    channel: IpcEvents,
    listener: (event: Electron.IpcMainInvokeEvent, ...args: any[]) => any,
  ) {
    ipcMain.handleOnce(channel, listener);
  }
}

export const ipcMainManager = new IpcMainManager();
