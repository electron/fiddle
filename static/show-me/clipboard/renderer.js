window.addEventListener('DOMContentLoaded', () => {
  const copyButton = document.querySelector('#copy')
  const pasteButton = document.querySelector('#paste')
  const textarea = document.querySelector('textarea')
  copyButton.onclick = () => {
    window.clipboard.writeText('Hello from Electron!')
  }
  pasteButton.onclick = async () => {
    textarea.value = await window.clipboard.readText()
  }
})
