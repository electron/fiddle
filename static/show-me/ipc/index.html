<!DOCTYPE html>
<html lang="en">
<head>
  <title>IPC Example</title>
</head>
<body>
  <h1>IPC Example</h1>
  <p>Open the developer tools to see the IPC in action.</p>
</body>
</html>
