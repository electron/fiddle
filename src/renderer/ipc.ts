import { EventEmitter } from 'events';

import { ipcRenderer } from 'electron';

import { IpcEvents, ipcRendererEvents } from '../ipc-events';

/**
 * The main purpose of this class is to be the central
 * gathering place for IPC calls the main process sends
 * or listens to.
 *
 * @class IpcManager
 * @extends {EventEmitter}
 */
class IpcRendererManager extends EventEmitter {
  constructor() {
    super();

    ipcRendererEvents.forEach((name) => {
      ipcRenderer.removeAllListeners(name);
      ipcRenderer.on(name, (...args: Array<any>) => this.emit(name, ...args));
    });
  }

  /**
   * Central method to send an IPC message to the main process.
   * Currently very simple, but centralized here so that future
   * refactors will be a lot easier.
   *
   * @param {IpcEvents} channel
   * @param {...Array<any>} args
   * @memberof IpcRendererManager
   */
  public send(channel: IpcEvents, ...args: Array<any>) {
    ipcRenderer.send(channel, ...args);
  }

  public invoke(channel: IpcEvents, ...args: Array<any>) {
    return ipcRenderer.invoke(channel, ...args);
  }
}

export const ipcRendererManager = new IpcRendererManager();
