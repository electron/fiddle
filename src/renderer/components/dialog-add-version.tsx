import * as React from 'react';

import * as path from 'path';

import {
  Button,
  Callout,
  Dialog,
  FileInput,
  InputGroup,
  Intent,
} from '@blueprintjs/core';
import { Installer } from '@vertedinde/fiddle-core';
import * as fs from 'fs-extra';
import { observer } from 'mobx-react';
import * as semver from 'semver';

import { Version } from '../../interfaces';
import { IpcEvents } from '../../ipc-events';
import { getElectronNameForPlatform } from '../../utils/electron-name';
import { ipcRendererManager } from '../ipc';
import { AppState } from '../state';
import { getLocalVersionForPath } from '../versions';

interface AddVersionDialogProps {
  appState: AppState;
}

interface AddVersionDialogState {
  isValidElectron: boolean;
  isValidVersion: boolean;
  existingLocalVersion?: Version;
  folderPath?: string;
  version: string;
}

/**
 * Build a default name for a local Electron version
 * from its dirname.
 *
 * @param {string} dirname
 * @return {string} human-readable local build name
 */
function makeLocalName(folderPath: string) {
  // take a dirname like '/home/username/electron/gn/main/src/out/testing'
  // and return something like 'gn/main - testing'
  const tokens = folderPath.split(path.sep);
  const buildType = tokens.pop(); // e.g. 'testing' or 'release'
  const leader = tokens
    // remove 'src/out/' -- they are in every local build, so make poor names
    .slice(0, -2)
    .join(path.sep)
    // extract about enough for the end result to be about 20 chars
    .slice(-20 + buildType!.length)
    // remove any fragment in case the prev slice cut in the middle of a name
    .split(path.sep)
    .slice(1)
    .join(path.sep);
  return `${leader} - ${buildType}`;
}

/**
 * The "add version" dialog allows users to add custom builds of Electron.
 *
 * @class AddVersionDialog
 * @extends {React.Component<AddVersionDialogProps, AddVersionDialogState>}
 */
export const AddVersionDialog = observer(
  class AddVersionDialog extends React.Component<
    AddVersionDialogProps,
    AddVersionDialogState
  > {
    constructor(props: AddVersionDialogProps) {
      super(props);

      this.state = {
        isValidVersion: false,
        isValidElectron: false,
        version: '',
      };

      this.onSubmit = this.onSubmit.bind(this);
      this.onClose = this.onClose.bind(this);
      this.onChangeVersion = this.onChangeVersion.bind(this);

      ipcRendererManager.on(
        IpcEvents.LOAD_LOCAL_VERSION_FOLDER,
        (_event, [file]) => {
          this.setFolderPath(file);
        },
      );
    }

    public componentWillUnmount() {
      ipcRendererManager.removeAllListeners(
        IpcEvents.LOAD_LOCAL_VERSION_FOLDER,
      );
    }

    /**
     * Handles a change of the file input
     *
     * @param {React.ChangeEvent<HTMLInputElement>} event
     */
    public setFolderPath(folderPath: string) {
      const isValidElectron = this.isValidElectronPath(folderPath);
      const existingLocalVersion = getLocalVersionForPath(folderPath);

      this.setState({ existingLocalVersion, folderPath, isValidElectron });
    }

    public onChangeVersion(event: React.ChangeEvent<HTMLInputElement>) {
      const version = event.target.value || '';
      const isValidVersion = !!semver.valid(version);

      this.setState({
        version,
        isValidVersion,
      });
    }

    /**
     * Verifies if the local electron path is valid
     *
     * @param {string} folderPath
     */
    public isValidElectronPath(folderPath: string) {
      const execPath = Installer.getExecPath(folderPath);
      return fs.existsSync(execPath);
    }

    /**
     * Handles the submission of the dialog
     *
     * @returns {Promise<void>}
     */
    public async onSubmit(): Promise<void> {
      const {
        folderPath,
        version,
        isValidElectron,
        existingLocalVersion,
      } = this.state;

      if (!folderPath) return;

      const toAdd: Version = {
        localPath: folderPath,
        version,
        name: makeLocalName(folderPath),
      };

      // swap to old local electron version if the user adds a new one with the same path
      if (isValidElectron && existingLocalVersion?.localPath) {
        // set previous version as active version
        this.props.appState.setVersion(existingLocalVersion.version);
      } else {
        this.props.appState.addLocalVersion(toAdd);
      }
      this.onClose();
    }

    /**
     * Closes the dialog
     */
    public onClose() {
      this.props.appState.isAddVersionDialogShowing = false;
      this.reset();
    }

    get buttons() {
      const {
        isValidElectron,
        isValidVersion,
        existingLocalVersion,
      } = this.state;
      const canAdd = isValidElectron && isValidVersion && !existingLocalVersion;
      const canSwitch = isValidElectron && existingLocalVersion;

      return [
        <Button
          icon="add"
          key="submit"
          disabled={!canAdd && !canSwitch}
          onClick={this.onSubmit}
          text={canSwitch ? 'Switch' : 'Add'}
        />,
        <Button
          icon="cross"
          key="cancel"
          onClick={this.onClose}
          text="Cancel"
        />,
      ];
    }

    public render() {
      const { isAddVersionDialogShowing } = this.props.appState;
      const inputProps = {
        onClick: (e: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
          e.preventDefault();
          ipcRendererManager.send(IpcEvents.SHOW_LOCAL_VERSION_FOLDER_DIALOG);
        },
      };
      const { folderPath } = this.state;

      const text =
        folderPath ||
        `Select the folder containing ${getElectronNameForPlatform()}...`;

      return (
        <Dialog
          isOpen={isAddVersionDialogShowing}
          onClose={this.onClose}
          title="Add local Electron build"
          className="dialog-add-version"
        >
          <div className="bp3-dialog-body">
            <FileInput
              id="custom-electron-version"
              inputProps={inputProps as any}
              text={text}
            />
            <br />
            {this.renderPath()}
          </div>
          <div className="bp3-dialog-footer">
            <div className="bp3-dialog-footer-actions">{this.buttons}</div>
          </div>
        </Dialog>
      );
    }

    private renderPath(): JSX.Element | null {
      const { isValidElectron, folderPath, existingLocalVersion } = this.state;
      const canSwitch = isValidElectron && existingLocalVersion;

      if (!folderPath) return null;
      return (
        <Callout>
          {this.buildDialogText()}
          {!canSwitch && this.renderVersionInput()}
        </Callout>
      );
    }

    private buildDialogText(): string {
      const { isValidElectron, existingLocalVersion } = this.state;
      const canSwitch = isValidElectron && existingLocalVersion;

      if (canSwitch)
        return `This folder is already in use as version "${
          existingLocalVersion!.version
        }". Would you like to switch to that version now?`;

      if (isValidElectron)
        return `We found an ${getElectronNameForPlatform()} in this folder.`;

      return `We did not find a ${getElectronNameForPlatform()} in this folder...`;
    }

    private renderVersionInput(): JSX.Element | null {
      const { isValidElectron, isValidVersion, version } = this.state;
      if (!isValidElectron) return null;

      return (
        <>
          <p>
            Please specify a version, used for typings and the name. Must be{' '}
            <code>semver</code> compliant.
          </p>
          <InputGroup
            intent={isValidVersion ? undefined : Intent.DANGER}
            value={version}
            onChange={this.onChangeVersion}
            placeholder="4.0.0"
          />
        </>
      );
    }

    /**
     * Reset this component's state
     */
    private reset(): void {
      this.setState({
        isValidElectron: false,
        isValidVersion: false,
        version: '',
        folderPath: undefined,
      });
    }
  },
);
