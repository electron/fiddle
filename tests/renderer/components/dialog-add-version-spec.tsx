import * as React from 'react';

import { shallow } from 'enzyme';

import { IpcEvents } from '../../../src/ipc-events';
import { AddVersionDialog } from '../../../src/renderer/components/dialog-add-version';
import { ipcRendererManager } from '../../../src/renderer/ipc';
import { StateMock } from '../../mocks/mocks';
import { overridePlatform, resetPlatform } from '../../utils';

jest.mock('../../../src/renderer/ipc');

describe('AddVersionDialog component', () => {
  let store: StateMock;

  const mockFile = '/test/file';

  beforeAll(() => {
    // We render the buttons different depending on the
    // platform, so let' have a uniform platform for unit tests
    overridePlatform('darwin');
  });

  afterAll(() => {
    resetPlatform();
  });

  beforeEach(() => {
    ({ state: store } = (window as any).ElectronFiddle.app);
  });

  it('renders', () => {
    const wrapper = shallow(<AddVersionDialog appState={store as any} />);

    wrapper.setState({
      isValidVersion: true,
      isValidElectron: true,
      folderPath: mockFile,
    });

    expect(wrapper).toMatchSnapshot();

    wrapper.setState({
      isValidVersion: false,
      isValidElectron: true,
      folderPath: mockFile,
    });

    expect(wrapper).toMatchSnapshot();

    wrapper.setState({
      isValidVersion: true,
      isValidElectron: true,
      existingLocalVersion: {
        version: '2.2.2',
        localPath: mockFile,
      },
      folderPath: mockFile,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('overrides default input with Electron dialog', () => {
    const preventDefault = jest.fn();

    const wrapper = shallow(<AddVersionDialog appState={store as any} />);
    const inp = wrapper.find('#custom-electron-version');
    inp.dive().find('input[type="file"]').simulate('click', { preventDefault });

    expect(ipcRendererManager.send as jest.Mock).toHaveBeenCalledWith(
      IpcEvents.SHOW_LOCAL_VERSION_FOLDER_DIALOG,
    );
    expect(preventDefault).toHaveBeenCalled();
  });

  describe('setFolderPath()', () => {
    it('does something', async () => {
      // (getIsDownloaded as jest.Mock).mockReturnValue(true);
      const wrapper = shallow(<AddVersionDialog appState={store as any} />);
      (wrapper.instance() as any).isValidElectronPath = () => true;
      await (wrapper.instance() as any).setFolderPath('/test/');

      expect(wrapper.state('isValidElectron')).toBe(true);
      expect(wrapper.state('folderPath')).toBe('/test/');
    });
  });

  describe('onChangeVersion()', () => {
    it('handles valid input', () => {
      const wrapper = shallow(<AddVersionDialog appState={store as any} />);

      (wrapper.instance() as any).onChangeVersion({
        target: { value: '3.3.3' },
      });
      expect(wrapper.state('isValidVersion')).toBe(true);
      expect(wrapper.state('version')).toBe('3.3.3');
    });

    it('handles invalid input', () => {
      const wrapper = shallow(<AddVersionDialog appState={store as any} />);

      (wrapper.instance() as any).onChangeVersion({ target: { value: 'foo' } });
      expect(wrapper.state('isValidVersion')).toBe(false);
      expect(wrapper.state('version')).toBe('foo');

      (wrapper.instance() as any).onChangeVersion({ target: {} });
      expect(wrapper.state('isValidVersion')).toBe(false);
      expect(wrapper.state('version')).toBe('');
    });
  });

  describe('onSubmit', () => {
    it('does not do anything without a file', async () => {
      const wrapper = shallow(<AddVersionDialog appState={store as any} />);

      await (wrapper.instance() as any).onSubmit();

      expect(store.addLocalVersion).toHaveBeenCalledTimes(0);
    });

    it('adds a local version using the given data', async () => {
      const wrapper = shallow(<AddVersionDialog appState={store as any} />);

      wrapper.setState({
        version: '3.3.3',
        folderPath: '/test/path',
      });

      await (wrapper.instance() as any).onSubmit();

      expect(store.addLocalVersion).toHaveBeenCalledTimes(1);

      const result = store.addLocalVersion.mock.calls[0][0];

      expect(result.localPath).toBe('/test/path');
      expect(result.version).toBe('3.3.3');
    });

    it('shows dialog warning when adding duplicate local versions', async () => {
      const wrapper = shallow(<AddVersionDialog appState={store as any} />);

      wrapper.setState({
        isValidElectron: true,
        folderPath: '/test/path',
        version: '3.3.3',
        existingLocalVersion: {
          version: '2.2.2',
          localPath: '/test/path',
        },
      });

      expect(wrapper).toMatchSnapshot();
    });
  });
});
