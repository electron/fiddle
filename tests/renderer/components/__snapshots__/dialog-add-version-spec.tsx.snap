// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddVersionDialog component onSubmit shows dialog warning when adding duplicate local versions 1`] = `
<Blueprint3.Dialog
  canOutsideClickClose={true}
  className="dialog-add-version"
  isOpen={false}
  onClose={[Function]}
  title="Add local Electron build"
>
  <div
    className="bp3-dialog-body"
  >
    <Blueprint3.FileInput
      hasSelection={false}
      id="custom-electron-version"
      inputProps={
        Object {
          "onClick": [Function],
        }
      }
      text="/test/path"
    />
    <br />
    <Blueprint3.Callout>
      This folder is already in use as version "2.2.2". Would you like to switch to that version now?
    </Blueprint3.Callout>
  </div>
  <div
    className="bp3-dialog-footer"
  >
    <div
      className="bp3-dialog-footer-actions"
    >
      <Blueprint3.Button
        disabled={false}
        icon="add"
        key="submit"
        onClick={[Function]}
        text="Switch"
      />
      <Blueprint3.Button
        icon="cross"
        key="cancel"
        onClick={[Function]}
        text="Cancel"
      />
    </div>
  </div>
</Blueprint3.Dialog>
`;

exports[`AddVersionDialog component renders 1`] = `
<Blueprint3.Dialog
  canOutsideClickClose={true}
  className="dialog-add-version"
  isOpen={false}
  onClose={[Function]}
  title="Add local Electron build"
>
  <div
    className="bp3-dialog-body"
  >
    <Blueprint3.FileInput
      hasSelection={false}
      id="custom-electron-version"
      inputProps={
        Object {
          "onClick": [Function],
        }
      }
      text="/test/file"
    />
    <br />
    <Blueprint3.Callout>
      We found an Electron.app in this folder.
      <p>
        Please specify a version, used for typings and the name. Must be
         
        <code>
          semver
        </code>
         compliant.
      </p>
      <Blueprint3.InputGroup
        onChange={[Function]}
        placeholder="4.0.0"
        value=""
      />
    </Blueprint3.Callout>
  </div>
  <div
    className="bp3-dialog-footer"
  >
    <div
      className="bp3-dialog-footer-actions"
    >
      <Blueprint3.Button
        disabled={false}
        icon="add"
        key="submit"
        onClick={[Function]}
        text="Add"
      />
      <Blueprint3.Button
        icon="cross"
        key="cancel"
        onClick={[Function]}
        text="Cancel"
      />
    </div>
  </div>
</Blueprint3.Dialog>
`;

exports[`AddVersionDialog component renders 2`] = `
<Blueprint3.Dialog
  canOutsideClickClose={true}
  className="dialog-add-version"
  isOpen={false}
  onClose={[Function]}
  title="Add local Electron build"
>
  <div
    className="bp3-dialog-body"
  >
    <Blueprint3.FileInput
      hasSelection={false}
      id="custom-electron-version"
      inputProps={
        Object {
          "onClick": [Function],
        }
      }
      text="/test/file"
    />
    <br />
    <Blueprint3.Callout>
      We found an Electron.app in this folder.
      <p>
        Please specify a version, used for typings and the name. Must be
         
        <code>
          semver
        </code>
         compliant.
      </p>
      <Blueprint3.InputGroup
        intent="danger"
        onChange={[Function]}
        placeholder="4.0.0"
        value=""
      />
    </Blueprint3.Callout>
  </div>
  <div
    className="bp3-dialog-footer"
  >
    <div
      className="bp3-dialog-footer-actions"
    >
      <Blueprint3.Button
        disabled={true}
        icon="add"
        key="submit"
        onClick={[Function]}
        text="Add"
      />
      <Blueprint3.Button
        icon="cross"
        key="cancel"
        onClick={[Function]}
        text="Cancel"
      />
    </div>
  </div>
</Blueprint3.Dialog>
`;

exports[`AddVersionDialog component renders 3`] = `
<Blueprint3.Dialog
  canOutsideClickClose={true}
  className="dialog-add-version"
  isOpen={false}
  onClose={[Function]}
  title="Add local Electron build"
>
  <div
    className="bp3-dialog-body"
  >
    <Blueprint3.FileInput
      hasSelection={false}
      id="custom-electron-version"
      inputProps={
        Object {
          "onClick": [Function],
        }
      }
      text="/test/file"
    />
    <br />
    <Blueprint3.Callout>
      This folder is already in use as version "2.2.2". Would you like to switch to that version now?
    </Blueprint3.Callout>
  </div>
  <div
    className="bp3-dialog-footer"
  >
    <div
      className="bp3-dialog-footer-actions"
    >
      <Blueprint3.Button
        disabled={false}
        icon="add"
        key="submit"
        onClick={[Function]}
        text="Switch"
      />
      <Blueprint3.Button
        icon="cross"
        key="cancel"
        onClick={[Function]}
        text="Cancel"
      />
    </div>
  </div>
</Blueprint3.Dialog>
`;
